tools/cldr-apps/src/org/unicode/cldr/web/DataSection.java

```diff
@@ -1424,6 +1429,42 @@ public boolean hadVotesSometimeThisRelease() {
         public boolean wouldInheritNull() {
             return inheritedValue == null;
         }
+
+        /**
+         * For annotations, include the root value as a candidate item that can't be voted for.
+         * This is so that people can search for, e.g., "E12" to find rows for emoji that are new.
+         */
+        public void addAnnotationRootValue() {
+            if (xpath.startsWith("//ldml/annotations/annotation")) {
+                CLDRFile rootFile = sm.getSTFactory().make(CLDRLocale.ROOT.getBaseName(), true);
```

Comment:
`populateFromThisXpath(String xpath)` is called on every path. I know make should be cached, but this seems like it could be a contention point.  Could this move to a getter on `DataSection` (next to the ubiquitous `sm`) `CLDRFile DataSection.getRootFile()` that calls this once? Or even just calculate it at DataSection construction?

Reply:
Good idea. I'll do a follow-up PR with that.